research/conditional/moe_layers/token_choice.py

```diff
@@ -41,6 +44,22 @@ def __init__(
         self.capacity_factor = capacity_factor
         self.expert_inner_function = expert_inner_function
         self.load_balancing_loss_weight = load_balancing_loss_weight
+        self.moe_values_exp = (
+            moe_values_exp
+            if moe_values_exp != -1
```

Comment:
why -1?

Reply:
it means it's trainable

Reply:
It's a weird convention tbh. Could we change that somehow?